src/sagemaker/pipeline.py

```diff
@@ -313,7 +313,7 @@ def register(
             if model.model_data is None:
                 raise ValueError("SageMaker Model Package cannot be created without model data.")
         if model_package_group_name is not None:
-            container_def = self.pipeline_container_def(inference_instances[0])
+            container_def = self.pipeline_container_def(inference_instances[0] if inference_instances else None)
```

Comment:
Can you also update the `instance_type` argument in `pipeline_container_def` and make its default value is None?
https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/pipeline.py#L87

Reply:
Done.